src/cli/commands/workspace.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import type Config from '../../config.js';
+import {MessageError} from '../../errors.js';
+import type {Reporter} from '../../reporters/index.js';
+import * as child from '../../util/child.js';
+import * as filter from '../../util/filter.js';
+
+export function setFlags() {}
+
+export function hasWrapper(): boolean {
+  return true;
+}
+
+export async function run(
+  config: Config,
+  reporter: Reporter,
+  flags: Object,
+  args: Array<string>,
+): Promise<void> {
+  const projectPath = await config.findProject(config.cwd);
+
+  if (!projectPath) {
+    throw new MessageError(reporter.lang('projectRootNotFound', config.cwd));
+  }
+
+  const manifest = await config.findManifest(projectPath);
+  const workspaces = await config.resolveWorkspaces(projectPath, manifest.workspaces);
+
+  if (args.length < 1) {
+    throw new MessageError(reporter.lang('projectMissingWorkspace'));
+  }
+
+  if (args.length < 2) {
+    throw new MessageError(reporter.lang('projectMissingCommand'));
+  }
+
+  const [workspaceName, ... rest] = args;
+
+  if (!Object.prototype.hasOwnProperty.call(workspaces, workspaceName)) {
+    throw new MessageError(reporter.lang('projectUnknownWorkspace', workspaceName));
+  }
+
+  try {
+    await child.spawn(process.argv[0], [process.argv[1], ... rest], { stdio: 'inherit' });
+  } catch (err) {
```

Comment:
The issue I had was that regular errors are already handled by the spawned child. Discarding the error here was to good way to prevent the error to be printed twice:

<img width="664" alt="screen shot 2017-05-17 at 11 08 09 am" src="https://cloud.githubusercontent.com/assets/1037931/26149165/2e2bb518-3af1-11e7-83a1-ecb572bf08ba.png">

However, it also swallows the exit code ... what would you think about manually calling `process.exit` with the exit code returned by the child?